src/addons/draggable/RowDragLayer.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component, PropTypes } from 'react';
+import { DragLayer } from 'react-dnd';
+import Selectors from '../data/Selectors';
+
+const layerStyles = {
+  cursor: '-webkit-grabbing',
+  position: 'fixed',
+  pointerEvents: 'none',
+  zIndex: 100,
+  left: 0,
+  top: 0,
+  width: '100%',
+  height: '100%'
+};
+
+function getItemStyles(props) {
+  const { currentOffset } = props;
+  if (!currentOffset) {
+    return {
+      display: 'none'
+    };
+  }
+
+  const { x, y } = currentOffset;
+  const transform = `translate(${x}px, ${y}px)`;
+  return {
+    transform: transform,
+    WebkitTransform: transform
+  };
+}
+
+class CustomDragLayer extends Component {
+
+  isDraggedRowSelected(selectedRows) {
+    let {item, rowSelection} = this.props;
+    if (selectedRows && selectedRows.length > 0) {
+      let key = rowSelection.selectBy.keys.rowKey;
+      return selectedRows.filter(r => r[key] === item.data[key]).length > 0;
+    }
+    return false;
+  }
+
+  getRows() {
+    let rows = [];
+    let {rowsCount, rowGetter} = this.props;
+    for (let j = 0; j < rowsCount; j++) {
+      rows.push(rowGetter(j));
+    }
+    return rows;
+  }
+
+  getDraggedRows() {
+    let draggedRows;
+    let {rowSelection} = this.props;
+    if (rowSelection && rowSelection.selectBy.keys) {
+      let rows = this.getRows();
+      let {rowKey, values} = rowSelection.selectBy.keys;
+      let selectedRows = Selectors.getSelectedRowsByKey({rowKey: rowKey, selectedKeys: values, rows: rows});
+      draggedRows = this.isDraggedRowSelected(selectedRows) ? selectedRows : [this.props.rowGetter(this.props.item.idx)];
+    } else {
+      draggedRows = [this.props.rowGetter(this.props.item.idx)];
+    }
+    return draggedRows;
+  }
+
+  renderDraggedRows() {
+    return this.getDraggedRows().map((r, i) => {
+      return <tr key={`dragged-row-${i}`}>{this.renderDraggedCells(r, i) }</tr>;
+    });
+  }
+
+  renderDraggedCells(item, rowIdx) {
+    let cells = [];
+    if (item != null) {
+      for (let c in item) {
```

Comment:
Shouldn't this be a map? Looks like the length for the original and result array are the same